provider.go

```diff
@@ -29,3 +29,16 @@ type Provider interface {
 	// Open retrieves the provider's secrets.
 	Open(ctx context.Context, inputs map[string]Value, executionContext EnvExecContext) (Value, error)
 }
+
+// A Rotator enables environments to rotate a secret.
+// Rotators persist internal state by writing back to the environment.
```

Comment:
should rephrase this doc comment along the lines of "it is the responsibility of the caller to appropriately persist rotation state (e.g. by writing it back to the environment definition).